src/periodic-table/Element.ts

```diff
@@ -0,0 +1,194 @@
+import { Element } from "./elements";
+
+enum Subshell {
+  s = "s",
+  p = "p",
+  d = "d",
+  f = "f",
+}
+
+const subshellMax = (subshell: Subshell): number => {
+  let ret = 0;
+  switch (subshell) {
+    case Subshell.s:
+      ret = 2;
+      break;
+    case Subshell.p:
+      ret = 6;
+      break;
+    case Subshell.d:
+      ret = 10;
+      break;
+    case Subshell.f:
+      ret = 14;
+      break;
+  }
+  return ret;
+};
+
+const intToSubshell = (n: number): Subshell => {
+  let ret: Subshell;
+  switch (n) {
+    case 0:
+      ret = Subshell.s;
+      break;
+    case 1:
+      ret = Subshell.p;
+      break;
+    case 2:
+      ret = Subshell.d;
+      break;
+    case 3:
+      ret = Subshell.f;
+      break;
+    default:
+      ret = intToSubshell(n % 4); // could throw error instead
+      return ret;
+  }
+};
+
+interface Orbitals {
+  [orbital: string]: number;
+}
+
+class Shell {
+  n: number;
+  orbitals: Orbitals;
+
+  constructor(n: number) {
+    this.n = n;
+    for (let i = 0; i < n && i < 4; i++) {
+      this.orbitals[intToSubshell(n)] = 0;
+    }
+  }
+
+  addElectrons(orbital: Subshell, electrons: number): number {
+    const maxInShell = subshellMax(orbital);
+    const shellElectrons = Math.min(electrons, maxInShell);
+    this.orbitals[orbital] = shellElectrons;
+    return electrons - shellElectrons;
+  }
+
+  remElectrons(orbital: Subshell, electrons: number): number {
+    const shellElectrons = this.orbitals[orbital];
+    const toRemove = Math.min(shellElectrons, electrons);
+    this.orbitals[orbital] = shellElectrons - toRemove;
+    return electrons - toRemove;
+  }
+
+  hasElectrons = (orbital: Subshell) => {
+    return this.orbitals[orbital] > 0;
+  };
+
+  toString(orbital: Subshell) {
+    return `${this.n}${orbital}${this.orbitals[orbital]}`;
+  }
+}
+
+const shells = [
+  "1s",
+  "2s",
+  "2p",
+  "3s",
+  "3p",
+  "4s",
+  "3d",
+  "4p",
+  "5s",
+  "4d",
+  "5p",
+  "6s",
+  "4f",
+  "5d",
+  "6p",
+  "7s",
+  "5f",
+  "6d",
+  "7p",
+];
+
+interface ElectronShells {
+  [shell: string]: Shell;
+}
+
+export const getShells = (e: Element, charge = 0) => {
```

Comment:
can you please add some unit tests for this logic? I'm not so familiar with orbitals so I'd like to see some true/false/edge cases. please use jest.